test/jdk/java/lang/ProcessBuilder/Basic.java

```diff
@@ -2647,14 +2617,31 @@ public void run() {
 
             long end = System.nanoTime();
             if ((end - start) < 500000000)
-                fail("Test failed: waitFor didn't take long enough");
+                fail("Test failed: waitFor didn't take long enough (" + (end - start) + "ns)");
 
             p.destroy();
 
             p.waitFor(1000, TimeUnit.MILLISECONDS);
         } catch (Throwable t) { unexpected(t); }
     }
 
+    /**
+     * Return the list of process arguments for a child to sleep 10 minutes (600000 milliseconds).
+     * @return A list of process arguments to sleep 10 minutes.
+     */
+    private static List<String> getSleepArgs() {
+        List<String> childArgs = null;
+        Path sleepExe = TEST_NATIVEPATH.resolve("sleepmillis");
+        if (sleepExe.toFile().canExecute()) {
```

Comment:
Why is the fallback necessary? Other test cases such as test/jdk/tools/launcher/JliLaunchTest.java do not have such a fallback.

Also, I noticed that JliLaunchTest does something like this:

```
        Path launcher = Paths.get(System.getProperty("test.nativepath"),
            "JliLaunchTest" + (Platform.isWindows() ? ".exe" : ""));
```

but test/jdk/java/lang/reflect/exeCallerAccessTest/CallerAccessTest.java doesn't add ".exe", so it may not be necessary.

Reply:
See above, the java Child is more portable and lower maintenance.
Windows looks for xxx.exe if xxx is not found.

Reply:
How about adding a comment to explain why the fallback is necessary?

Also, if `jtreg -nativepath:xxx` is specified, but there's no xxx/sleepmillis, or xxx/sleepmillis is not executable, that should be a setup error (or a bug in the test itself). E.g., if we are testing inside mach5, we should always execute the native program, and should not silently fallback to the java Child program. Otherwise, setup problems in mach5 might bring us back to the mysterious intermittent failure.

(The current version of the code is buggy on Windows and will always silently fall back to Child because the executable is named "sleepmillis.exe", not "sleepmillis").